suite-common/wallet-core/src/fiat-rates/fiatRatesThunks.ts

```diff
@@ -0,0 +1,364 @@
+    (_, { dispatch, extra, getState }) => {
+        const {
+            selectors: { selectEnabledNetworks },
+        } = extra;
+        const enabledNetworks = selectEnabledNetworks(getState());
+        const fiat = selectCoins(getState());
+        fiat.forEach(f => {
+            const rateNetwork = (f.mainNetworkSymbol ?? f.symbol) as Network['symbol'];
+            if (!enabledNetworks.includes(rateNetwork)) {
+                dispatch(fiatRatesActions.removeFiatRate(f));
+            }
+        });
+    },
+);
+
+/**
+ *  Returns an array with coin tickers that need fullfil several conditions.
+ *  Array of coin tickers is combined from 2 sources - config file and fiat reducer itself
+ *  Conditions:
+ *  1. network for a given ticker needs to be enabled
+ *  2a. no rates available yet (first fetch)
+ *  OR
+ *  2b. duration since the last check is greater than passed `interval`
+ *  Timestamp is extracted via `timestampFunc`.
+ *
+ */
+export const getFiatStaleTickersThunk = createThunk(
+    `${actionPrefix}/getStaleTickers`,
+    (
+        { timestampFunc, interval, includeTokens }: GetFiatStaleTickersThunkPayload,
+        { extra, getState },
+    ) => {
+        const {
+            selectors: { selectEnabledNetworks, selectBlockchain },
+        } = extra;
+        const enabledNetworks = selectEnabledNetworks(getState());
+        const fiat = selectCoins(getState());
+        const blockchain = selectBlockchain(getState());
+
+        const watchedCoinTickers = FIAT.tickers
+            .filter(t => enabledNetworks.includes(t.symbol))
+            // use only connected backends
+            .filter(t => blockchain[t.symbol].connected);
+
+        const needUpdateFn = (t: TickerId) => {
+            // if no rates loaded yet, load them;
+            if (fiat.length === 0) return true;
+            const alreadyWatchedTicker = fiat.find(f => f.symbol === t.symbol);
+            // is not in fiat[], means is not watched, for example coin was added in settings, add it
+            if (!alreadyWatchedTicker) return true;
+
+            const timestamp = timestampFunc(alreadyWatchedTicker);
+            if (!timestamp) return true;
+            // otherwise load only older ones
+            return Date.now() - timestamp > interval;
+        };
+
+        const tickersToUpdate: TickerId[] = [];
+        watchedCoinTickers
+            .filter(needUpdateFn)
+            .forEach(t => tickersToUpdate.push({ symbol: t.symbol }));
+
+        if (includeTokens) {
+            // use only tokens which mainNetworkSymbol are assigned to watchedCoinTickers
+            const tokenTickers = fiat.filter(
+                t =>
+                    t.mainNetworkSymbol &&
+                    watchedCoinTickers.find(w => w.symbol === t.mainNetworkSymbol),
+            );
+            tokenTickers.filter(needUpdateFn).forEach(t => tickersToUpdate.push(t));
+        }
+
+        return tickersToUpdate;
```

Comment:
please verify all places where this thunk is called, if it counts with return value wrapped in action. I seen some places where some other thunks return type was overrided to `any`.

Reply:
I am waiting for the promise to resolve using `.unwrap()` (https://github.com/trezor/trezor-suite/pull/6118/files#diff-e941674440edbc0538d5d356706ca744d9f81c79be7cd0f1ce79c5f7a28352ddR196). Then you get the correct type:
![Snímek obrazovky 2022-08-29 v 19 41 02](https://user-images.githubusercontent.com/2227592/187264551-bc0b067e-6b24-40c4-abb8-be38839cd4fe.png)

Reply:
Cool, so this only place where it was used?

Reply:
There is one more which is handled the same way as mentioned above - https://github.com/trezor/trezor-suite/pull/6118/files#diff-e941674440edbc0538d5d356706ca744d9f81c79be7cd0f1ce79c5f7a28352ddR246